shared/src/main/scala/mlscript/JSBackend.scala

```diff
@@ -740,12 +747,13 @@ class JSBackend(allowUnresolvedSymbols: Boolean) {
   }
 
   protected def declareNewTypeDefs(typeDefs: Ls[NuTypeDef]):
-    (Ls[TraitSymbol], Ls[NewClassSymbol], Ls[MixinSymbol], Ls[ModuleSymbol], HashMap[String, Ls[Term]]) = {
+    (Ls[TraitSymbol], Ls[NewClassSymbol], Ls[MixinSymbol], Ls[ModuleSymbol], HashMap[String, Ls[Term]], HashMap[String, Ls[Term]]) = {
     val traits = new ListBuffer[TraitSymbol]()
     val classes = new ListBuffer[NewClassSymbol]()
     val mixins = new ListBuffer[MixinSymbol]()
     val modules = new ListBuffer[ModuleSymbol]()
     val superParameters = HashMap[String, Ls[Term]]()
+    val initStmts = HashMap[String, Ls[Term]]()
```

Comment:
Can you explain why we need this `initStmts` hash map? It seems like *very* smelly code. Would be much better and more maintainable to just tore the statements where they are needed.

Reply:
Maybe I can store them in the corresponding symbols(e.g. `NewClassSymbol`)?

Reply:
Removed two `HashMap` and put everything we need into `*Symbol`.